### ISSUE TITLE
1.2.0-rc.2 stream rules problem with facility

### ISSUE THREAD

**Issue by razvanphp** [Graylog2/graylog2-server#1396]
We currently have two graylog servers, one with the new 1.2.0-rc.2 version and the old one with v1.1.6.

Both of them have the same streams, because the old one forwards through an output to the new one, and one of the streams gets more messages than it should on the new RC version only.

The rules configured for this particular stream are:
- ctxt_stream must match exactly feed
- facility must match exactly shop.live

The wrong behaviour is that the stream receives messages with only facility condition true, but without "ctxt_stream" attribute to be even present. If I remove the facility rule it seems to be fine.

Can you replicate this bug? Otherwise tell me how we can debug this, because on 1.1.6 are routed only the correct messages.

Thank you!

**Comment by edmundoa** [comment]
Hi,

On version 1.2 we introduced the possibility of doing OR operations on streams. Could you please indicate if you selected "match all rules" or "match any rules" on the edit stream rules page?

**Comment by razvanphp** [comment]
It is the default, to match all.

`Must match all of the 2 configured stream rule(s).`

Testing one message that was wrongly routed in this streams says that "it will not be routed to this stream" and that only the facility rule matches. If it makes any difference, those messages were also routed in another stream based on the source, and this is correct.

**Comment by joschi** [comment]
@razvanphp Could you please try to retrieve the stream rules from MongoDB and post them here? You can retrieve them with the following query using a MongoDB client (e. g. `mongo`):
```
db.streamrules.find(, { "stream_id": ObjectId("$STREAM_ID") });
```
Please replace `$STREAM_ID` in this query with the actual ID of the stream (e. g. `552b92b2e4b0c055e41ffb9e`).

**Comment by razvanphp** [comment]
As I said, it looks like it's doing OR instead of AND, in the web interface it appears correct, here is some evidences:

````
> db.streamrules.find({ "stream_id": ObjectId("55ea316be4b01a15684022c6") });
{ "_id" : ObjectId("55ea3185e4b01a15684022e1"), "field" : "ctxt_stream", "stream_id" : ObjectId("55ea316be4b01a15684022c6"), "type" : 1, "inverted" : false, "value" : "masterfeed" }
{ "_id" : ObjectId("55ed4a1ee4b01a1568437dc1"), "field" : "facility", "stream_id" : ObjectId("55ea316be4b01a15684022c6"), "type" : 1, "inverted" : false, "value" : "collins.aboutyou.shop.live" }
>
````

![screen shot 2015-09-07 at 10 39 22](https://cloud.githubusercontent.com/assets/4599319/9712721/e8c64dc0-554c-11e5-84b5-ddc512291c52.png)
![screen shot 2015-09-07 at 10 30 26](https://cloud.githubusercontent.com/assets/4599319/9712731/fee379de-554c-11e5-80e6-aecc776a2d65.png)

As you see in the last screenshot, the message matches the "facility" rule, but the "ctxt_stream" field is not even present in the message, but it still gets routed to this stream.

**Comment by bernd** [comment]
This got fixed in #1397 and will be released in the upcoming 1.2.0-rc.3 shortly. Please re-open this ticket if you still have problems with rc.3.

Thank you for the report!
